### PR TITLE
v3: NFT (Image Tracking) + clean code + doc enhancement

### PR DESCRIPTION
**Pull request by kalwalt**
**⚠️ All PRs have to be done versus 'dev' branch, so be aware of that, or we'll close your issue ⚠️**

**What kind of change does this PR introduce?**

From #674 (comment) #674 and #705
This PR want to add NFT feature  from [artoolkitx/jsartoolkit5](https://github.com/artoolkitx/jsartoolkit5) and also other changes/improvements

### **🖼 This will eventually make AR.js capable of detect and track 2D images.**

https://twitter.com/nicolocarp/status/1204841439160684549

Could be raise some bugs and incompatibilties between the code. 
**Can it be referenced to an Issue? If so what is the issue # ?**
From jsartoolkit5 repo:  PR  https://github.com/kalwalt/jsartoolkit5/pull/1
From AR.js issues:

* https://github.com/jeromeetienne/AR.js/issues/544
* https://github.com/jeromeetienne/AR.js/issues/149
* https://github.com/jeromeetienne/AR.js/issues/503
* https://github.com/jeromeetienne/AR.js/issues/420
* https://github.com/jeromeetienne/AR.js/issues/227
* https://github.com/jeromeetienne/AR.js/issues/190
* https://github.com/jeromeetienne/AR.js/issues/185

also, a repo cleaning will help for: https://github.com/jeromeetienne/AR.js/issues/433
and also: #722 

**How can we test it?**
Just test the examples and tell us what happens...

**Summary**

see the other PR:
#722 
#740 

**Does this PR introduce a breaking change?**

Current repository will continue to be up, with marker + gps support.

A new repository will be created on AR.js new organization, that will feature two libraries:
* one with marker+gps (as current)
* one with image tracking + gps

Because of incompatibility between markers and NFT.

**Please TEST your PR before proposing it. Specify here what device you have used for tests, version of OS and version of Browser**

**Other information**
I uploaded the new jsartookit5 libs and build the AR.js libs with makefile. 

**Progress**
- [x] Clean repo before starting the porting (removing old aruco and tango code)
- [x] Import new jsartoolkit5 version, coming with NFT
- [x] Adapt the code (threejs side) to support NFT (emulate what's already done for fiducial markers)
- [x] Fix the rendering of content on AR.js side, for NFT (onMarkerFound function)
- [x] Make it work on mobile
- [x] Enhance resolution in both mobile and desktop
- [x] Check if animations work with NFT. If not, or rendering rate is too high, use interpolation
- [x] Fix broken pattern marker (content appears shifted after NFT modifies, or maybe even before during the cleaning of the repo)
- [x] Remove all rendering code on the Worker
- [x] Adapt the code (aframe side) to support NFT (emulate what's already done for fiducial markers)
- [x] Create at least one example for threejs + NFT
- [x] Create at least one example for aframe + NFT
- [ ] Change documentation (principal Readme, also with NFT Marker generator + Readmes inside threejs + aframe folders)

### CODE REVIEW

File: aframe/examples/marker-events.html [hunk 1]
```diff
@@ -36,7 +36,7 @@
 	<br/>
 	Contact me any time at <a href='https://twitter.com/jerome_etienne' target='_blank'>@jerome_etienne</a>
 </div>
-	<a-scene embedded arjs='sourceType: webcam; detectionMode: mono_and_matrix; matrixCodeType: 3x3; sourceWidth:1280; sourceHeight:960; displayWidth: 1280; displayHeight: 960;'>
```
Comment (kalwalt) [inline aframe/examples/marker-events.html]: i need to remove this line because maybe my Mobile device do not support this...

**Comment by kalwalt** [comment]
I started to remove aruco in small pieces, in this way is easier to identify breaking changes.

File: aframe/examples/marker-events.html (hunk 1 above)
Comment (nicolocarpignoli) [reply aframe/examples/marker-events.html]: it absolutely makes sense, but please remove only this part:

` sourceWidth:1280; sourceHeight:960; displayWidth: 1280; displayHeight: 960`

Comment (kalwalt) [reply aframe/examples/marker-events.html]: yes it's like that, i deleted only that piece of code i have only commented on the left side,

Comment (kalwalt) [reply aframe/examples/marker-events.html]: Anyway it's strange because it's only the double of 640 x 480.

**Comment by nicolocarpignoli** [comment]
is all the tango/aruco code removed?

**Comment by kalwalt** [comment]
Not completely but it will be soon.

**Comment by kalwalt** [comment]
aruco code is completely removed, and testing examples are working correctly, ready to add NFT!

**Comment by kalwalt** [comment]
In the nft.html threejs example the NFT marker is tracked but i can not see the Flamingo model. We need to check this...

**Comment by kalwalt** [comment]
I can see that there are chunks of aruco code in threex-artoolkitcontext-nft.js to be removed and maybe in other parts.

**Comment by nicolocarpignoli** [comment]
Ok, first it's better to remove ALL aruco (you can search for 'aruco' and 'tango' strings in the project) and then we can try to add NFT.html. I can help with adding it.

**Comment by kalwalt** [comment]
> Ok, first it's better to remove ALL aruco (you can search for 'aruco' and 'tango' strings in the project) and then we can try to add NFT.html. I can help with adding it.

I hope to work on it this afternoon !

**Comment by kalwalt** [comment]
With the latest commit https://github.com/jeromeetienne/AR.js/pull/742/commits/97be36b2a8ef4d899600940be720fa777d85a613 The Flamingo model is shown onto the NFT marker! Actually pattern marker (threejs and aframe) and NFT marker with threejs both works toghether!! Next step is to re-add NFT to aframe.... we are very close really to a release. :smile: 
I think that probably we can remove part of the code in the threex-...-nft that is not necessary (for example about new white region and other) we can discuss this of course.

File: three.js/src/threex/threex-artoolkitcontext-nft.js [hunk 2]
```diff
@@ -29,13 +29,18 @@ ARjs.Context = THREEx.ArToolkitContext = function (parameters) {
         // the patternRatio inside the artoolkit marker - artoolkit only
         patternRatio: 0.5,
 
+        // Labeling mode for markers - ['black_region', 'white_region']
```
Comment (kalwalt) [inline three.js/src/threex/threex-artoolkitcontext-nft.js]: we don't need this and also other part of the code, but i'm not sure if we can remove them. Maybe better to do in another PR.

File: three.js/src/threex/threex-artoolkitcontext-nft.js [hunk 3]
```diff
@@ -131,13 +136,15 @@ ARjs.Context.prototype.update = function (srcElement) {
 
     // mark all markers to invisible before processing this frame
     this._arMarkersControls.forEach(function (markerControls) {
-        markerControls.object3d.visible = false
+        if (!markerControls.context.arController.showObject) {
```
Comment (kalwalt) [inline three.js/src/threex/threex-artoolkitcontext-nft.js]: i think that part missed code was causing to not show the model. An important teminder!

File: three.js/src/threex/threex-artoolkitcontext-nft.js (hunk 2 above)
Comment (nicolocarpignoli) [reply three.js/src/threex/threex-artoolkitcontext-nft.js]: we can do also on this PR but before we have to reach a stable state where everything works, marker and nft in aframe and three.js.

we have to remember about this.

**Comment by nicolocarpignoli** [comment]
This is it. We think to have reached a good release. We are finishing tests and then we will prepare several documentation enhancement. Should not take too much.
Also, this repository will be kept up but will be deprecated in favour of a new repository on new AR.js org.

This will solve also:

https://github.com/jeromeetienne/AR.js/issues/190
https://github.com/jeromeetienne/AR.js/issues/469
https://github.com/jeromeetienne/AR.js/issues/253

**Comment by nicolocarpignoli** [comment]
I will manually close all related issues.

The new v3 with Image Tracking will be available on the new AR.js repository, under AR.js organization.

I will close this PR and as soon as the v3 is ready, I will make a redirect using the principal Readme on this repository. Thank you all.
